core/src/com/unciv/logic/automation/unit/WorkerAutomation.kt

```diff
@@ -373,84 +412,59 @@ class WorkerAutomation(
     }
 
     /**
-     * Looks for work connecting cities
-     * @return whether we actually did anything
+     * Most importantly builds the cache so that [chooseImprovement] knows later what tiles a road should be built on
+     * Returns a list of all the cities close by that this worker may want to connect
      */
-    private fun tryConnectingCities(unit: MapUnit, minPriority: Float): Boolean {
-        if (bestRoadAvailable == RoadStatus.None || citiesThatNeedConnecting.isEmpty()) return false
-        val maxDistanceWanted = when {
-            minPriority > 4 -> -1
-            minPriority > 3 -> 0
-            minPriority > 2 -> 1
-            minPriority > 1 -> 2
-            minPriority > 0 -> 10
-            else -> 20
-        }
-        if (maxDistanceWanted < 0) return false
-
-        // Since further away cities take longer to get to and - most importantly - the canReach() to them is very long,
-        // we order cities by their closeness to the worker first, and then check for each one whether there's a viable path
-        // it can take to an existing connected city.
-        val candidateCities = citiesThatNeedConnecting.asSequence().filter {
+    private fun getNearbyCitiesToConnect(unit: MapUnit): List<City> {
+        if (bestRoadAvailable == RoadStatus.None || citiesThatNeedConnecting.isEmpty()) return listOf()
+        val candidateCities = citiesThatNeedConnecting.filter {
             // Cities that are too far away make the canReach() calculations devastatingly long
             it.getCenterTile().aerialDistanceTo(unit.getTile()) < 20
         }
-        if (candidateCities.none()) return false // do nothing.
+        if (candidateCities.none()) return listOf() // do nothing.
 
-        val isCandidateTilePredicate: (Tile) -> Boolean = { it.isLand && unit.movement.canPassThrough(it) }
+        // Search through ALL candidate cities to build the cache
+        for (toConnectCity in candidateCities) {
+            getRoadConnectionBetweenCities(unit, toConnectCity).filter { it.getUnpillagedRoad() < bestRoadAvailable }
```

Comment:
What's with the filter? It doesn't actually do anything

Reply:
Copy and paste. I don't know, I would have to look into it more.